torch/_export/serialize.py

```diff
@@ -46,30 +69,441 @@ def _reverse_map(d):
+    operator.floordiv,
+    operator.mod,
+}
+
+def export_sym_int(s: Union[int, torch.SymInt]) -> SymInt:
     if isinstance(s, int):
         return SymInt.create(as_int=s)
     elif isinstance(s, torch.SymInt):
-        return SymInt.create(as_symbol=str(s))
+        if is_concrete_int(s):
+            return SymInt.create(as_int=int(s))
+        else:
+            return SymInt.create(as_symbol=str(s))
     else:
-        raise ValueError(str(s))
+        raise SerializeError(
+            f"SymInt should be either symbol or int, got `{s}` of type `{type(s)}`"
+        )
 
 
-def _extract_tensor_meta(result: torch.Tensor) -> TensorMeta:
+def export_tensor_meta(t: torch.Tensor) -> TensorMeta:
     """
-    Extract a TensorMeta describing `result`.
+    Extract a TensorMeta describing `t`.
     """
     return TensorMeta(
-        dtype=_SCALAR_TYPES[result.dtype],
-        sizes=[_extract_sym_int(s) for s in result.shape],
-        requires_grad=result.requires_grad,
-        device=Device(type=result.device.type, index=result.device.index),
-        strides=[_extract_sym_int(s) for s in result.stride()],
+        dtype=_TORCH_TO_SERIALIZE_DTYPE[t.dtype],
+        sizes=[export_sym_int(s) for s in t.shape],
+        requires_grad=t.requires_grad,
+        device=Device(type=t.device.type, index=t.device.index),
+        strides=[export_sym_int(s) for s in t.stride()],
         storage_offset=0,
-        layout=_LAYOUTS[result.layout],
+        layout=_TORCH_TO_SERIALIZE_LAYOUT[t.layout],
     )
 
 
+def export_metadata(node: torch.fx.Node) -> Dict[str, str]:
+    ret = {}
+    if stack_trace := node.meta.get("stack_trace"):
+        ret["stack_trace"] = stack_trace
+    module_fqn = node.meta.get("module_fqn")
+    # Need an explicit None check instead of walrus operator, because
+    # module_fqn can be the empty string if the node belongs to the root.
+    # The walrus operator returns False on an empty string :(
+    if module_fqn is not None:
+        ret["module_fqn"] = module_fqn
+    # TODO(angelayi) add nn_module_stack and source_fn
+    return ret
+
+
+def export_operator(target, version) -> Operator:
+    if isinstance(target, str):
+        return Operator(name=target, version=version)
+    elif target in _SYM_INT_OPS:
+        return Operator(name=f"{target.__module__}.{target.__name__}", version=version)
+    elif isinstance(target, torch._ops.HigherOrderOperator):
+        return Operator(name=target.__name__, version=version)
+    else:
+        return Operator(name=str(target), version=version)
+
+
+def export_pytree(spec: pytree.TreeSpec) -> str:
+    # TODO(angelayi): spec
+    # assert isinstance(spec, pytree.TreeSpec), type(spec)
+    # dummy_leaves = [0] * spec.num_leaves
+    # tree = torch.utils._pytree.tree_unflatten(dummy_leaves, spec)
+    # _, tree = ex_pytree.tree_flatten(tree)
+    # return tree.to_str()
+    return ""
+    # TODO(angelayi): spec
+    return ""
+
+
+def export_signature(sig) -> GraphSignature:
```

Comment:
We should probably consolidate GraphSignature with ExportMetaData fields.

Reply:
Yes...this part is all tbd until ExportedProgram is figured out (cc @gmagogsfm)